lib/FrontendTool/FrontendTool.cpp

```diff
@@ -686,7 +686,13 @@ static void countStatsPostSema(UnifiedStatsReporter &Stats,
   }
 
   for (auto SF : Instance.getPrimarySourceFiles()) {
-    if (auto *R = SF->getReferencedNameTracker()) {
+    ReferencedNameTracker *Tracker = nullptr;
+    if (AST.LangOpts.DisableRequestBasedIncrementalDependencies) {
+      Tracker = SF->getReferencedNameTracker();
+    } else {
+      Tracker = SF->getRequestBasedReferencedNameTracker();
+    }
```

Comment:
I know this is standard C++, but it grates on my how it precludes the use of a ref or a *const . Would it be OK to use the ?: operator to enable the constifaction?

Reply:
Also this same code is repeated at line 951. Would be nice to factor it out.